work/v1alpha1/types.go

```diff
@@ -0,0 +1,150 @@
+// +kubebuilder:printcolumn:name="Applied",type="string",JSONPath=".status.conditions[?(@.type==\"ManifestworkApplied\")].status",description="Applied"
+
+// PlaceManifestWork is the Schema for the PlaceManifestWorks API. This custom resource is able to apply
+// ManifestWork using Placement for 0..n ManagedCluster(in their namespaces). It will also remove the ManifestWork custom resources
+// when deleted. Lastly the specific ManifestWork custom resources created per ManagedCluster namespace will be adjusted based on PlacementDecision
+// changes.
+type PlaceManifestWork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec reperesents the desired ManifestWork payload and Placement reference to be reconciled
+	Spec PlaceManifestWorkSpec `json:"spec,omitempty"`
+
+	// Status represent the current status of Placing ManifestWork resources
+	Status PlaceManifestWorkStatus `json:"status,omitempty"`
+}
+
+// PlaceManifestWorkSpec defines the desired state of PlaceManifestWork
+type PlaceManifestWorkSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// ManifestWorkSpec is the ManifestWorkSpec that will be used to generate a per-cluster ManifestWork
+	ManifestWorkTemplate work.ManifestWorkSpec `json:"manifestWorkTemplate"`
+
+	// PacementRef is the name of the Placement resource, from which a PlacementDecision will be found and used
+	// to distribute the ManifestWork
+	PlacementRef LocalPlacementReference `json:"placementRef,omitempty"`
+}
+
+// PlaceManifestWorkStatus defines the observed state of PlaceManifestWork
+type PlaceManifestWorkStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Conditions contains the different condition statuses for distrbution of ManifestWork resources
+	// Valid condition types are:
+	// 1. AppliedManifestWorks represents ManifestWorks have been distributed as per placement All, Partial, None, Problem
+	// 2. PlacementRefValid
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// ManifestWorkDelivery tracks each ManifestWork that was created,
+	// including that it was applied and its overall status
+	PlacedManifestWork []PlacedManifestWork `json:"placedManifestWork,omitempty"`
+
+	// Summary that reflects all relevant ManifestWorks
+	PlacedManifestWorkSummary []PlacedManifestWorkSummary `json:"summary"`
+}
+
+// localPlacementReference is the name of a Placement resource in current namespace
+type LocalPlacementReference struct {
+	// Name of the Placement resource in the current namespace
+	Name string `json:"name"`
+}
+
+type PlacedManifestWorkSummary struct {
+	Total       int `json:"total"`
```

Comment:
still needs doc/comment for each field here. Is it total based on manifest or work?

Reply:
done.